src/pybind/mgr/dashboard_v2/frontend/src/app/shared/components/table/table.component.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  Component, EventEmitter, OnInit, Input, Output, ViewChild, OnChanges, ComponentFactoryResolver, Type
+} from '@angular/core';
+import {DatatableComponent, TableColumn} from '@swimlane/ngx-datatable';
+import {TableDetailsDirective} from './table-details.directive';
+
+@Component({
+  selector: 'cd-table',
+  templateUrl: './table.component.html',
+  styleUrls: ['./table.component.scss']
+})
+export class TableComponent implements OnInit, OnChanges {
+  @ViewChild(DatatableComponent) table: DatatableComponent;
+  @ViewChild(TableDetailsDirective) detailTemplate: TableDetailsDirective;
+
+  @Input() data: any[]; // This is the array with the items to be shown
+  @Input() columns: TableColumn[]; // each item -> { prop: 'attribute name', name: 'display name' }
+  @Input() detailsComponent?: string; // name of the component fe 'TableDetailsComponent'
+  @Input() header? = true;
+
+  @Output() fetchData = new EventEmitter(); // Should be the function that will update the input data
+
+  search = '';
+  rows = [];
+  temp = [];
+  selected = [];
+  paginationClasses = {
+    pagerLeftArrow: 'i fa fa-angle-double-left',
+    pagerRightArrow: 'i fa fa-angle-double-right',
+    pagerPrevious: 'i fa fa-angle-left',
+    pagerNext: 'i fa fa-angle-right'
+  };
+  limit = 10;
+
+  constructor(private componentFactoryResolver: ComponentFactoryResolver) {
+  }
+
+  ngOnInit() {
+    this.reloadData();
+  }
+
+  ngOnChanges(changes) {
+    if (changes.data.firstChange) {
```

Comment:
Do you really need this `if`?

On my local environment this is preventing the display of the data that is received from a server request.

Reply:
Yeah this could be the case if data is "undefined" beforehand, didn't had that case in mind, thanks for spotting this.